internal/xdg/trashdir.go

```
@@ -238,42 +239,41 @@ func getAllMountpoints() ([]string, error) {
 	return mountpoints, nil
 }
 
-var mountinfo_GetMounts = mountinfo.GetMounts
+var mountinfo_Mounted = mountinfo.Mounted
+var realpath_Realpath = realpath.Realpath
 
-// Obtain a mount point associated with a file
+// Obtain a mount point associated with a file.
 // Same as df <PATH>
 func getMountpoint(path string) (string, error) {
-	// get mountpoints from /proc/self/mountinfo on Linux
-	// getfsstat(2) used on Mac (BSD)
-	mountpoints, err := mountinfo_GetMounts(nil)
+
+	// iterate over the parents of the real (without symlinks) path until we find a mount point
+
+	candidate, err := realpath_Realpath(path)
 	if err != nil {
 		return "", err
 	}
 
-	mountpoint := path
-OUTER:
 	for {
 		// root is always mounted
-		if mountpoint == string(os.PathSeparator) {
+		if candidate == string(os.PathSeparator) {
 			slog.Debug("root mountpoint is detected", "path", path)
-			break OUTER
+			break
 		}
 
-		if mountpoint == "." {
+		if candidate == "." {
 			// should not reached here
 			// check to prevent busy loop
 			return "", errors.New("mountpoint is '.'")
 		}
 
-		for _, m := range mountpoints {
-			if m.Mountpoint == mountpoint {
-				break OUTER
-			}
+		if mounted, err := mountinfo_Mounted(candidate); err == nil && mounted {
```

I think an err check is needed
(If for some reason mountpoints could not be taken or something unlitely. It is better to return an error and exit early)

---

That seems sensible.